src/zod/routers/nextRouter.ts

```diff
@@ -0,0 +1,38 @@
+import { ZodType, ZodTypeDef } from 'zod'
+import { PaginationQuery, AbstractQuery } from '../../engine'
+import {
+  useQuery as useQueryVanilla,
+  usePagination as usePaginationVanilla,
+  useQueryAndPagination as useQueryAndPaginationVanilla,
+} from '../../routers/nextRouter'
+import { zodParser } from '../util'
+import { NonNullableRecord } from '../../utils'
+
+export const useQuery = <
+  TQuery extends NonNullableRecord<TSchemaOutput>,
+  TSchemaOutput extends Partial<AbstractQuery>,
+  TSchemaDef extends ZodTypeDef = ZodTypeDef,
+  TSchemaInput = TSchemaOutput
+>(
+  defaultQuery: TQuery,
+  schemaQuery: ZodType<TSchemaOutput, TSchemaDef, TSchemaInput>
+) => useQueryVanilla(defaultQuery, zodParser(schemaQuery))
+
+export const useQueryAndPagination = <
+  TQuery extends NonNullableRecord<TSchemaOutput>,
+  TSchemaOutput extends Partial<AbstractQuery>,
+  TSchemaDef extends ZodTypeDef = ZodTypeDef,
+  TSchemaInput = TSchemaOutput
+>(
+  defaultQuery: TQuery,
+  schemaQuery: ZodType<TSchemaOutput, TSchemaDef, TSchemaInput>,
+  defaultPagination?: PaginationQuery
```

Comment:
I think the `NextRouterOptions` are missing here and in the `useQuery` hook.

`options?: Partial<AbstractQueryOptions & NextRouterOptions>`